app/main.py

```diff
@@ -1,2 +1,55 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.threshold = int(self.capacity * 2 / 3)
+        self.buckets = [[] for _ in range(self.capacity)]
+        self.length = 0
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.length == self.threshold:
+            self.resize_buckets()
+        hashed_key = hash(key)
+        index = hashed_key % self.capacity
+        while True:
+
+            if not self.buckets[index]:
```

Comment:
You may move call to `self.resize_buckets()` here, since this is the only case where it's actually needed.